aws/resource_aws_sns_sms_preferences.go

```diff
@@ -0,0 +1,159 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSnsSmsPreferences() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSnsSmsPreferencesSet,
+		Read:   resourceAwsSnsSmsPreferencesGet,
+		Update: resourceAwsSnsSmsPreferencesSet,
+		Delete: resourceAwsSnsSmsPreferencesDelete,
+
+		Schema: map[string]*schema.Schema{
+			"monthly_spend_limit": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"delivery_status_iam_role_arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"delivery_status_success_sampling_rate": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"default_sender_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"default_sms_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"usage_report_s3_bucket": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+const resourceId = "aws_sns_sms_id"
+
+var SMSAttributeMap = map[string]string{
```

Comment:
no need to export this I don't think? (same with the var below as well)

Reply:
Indeed, corrected